Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4142,9 +4144,149 @@ - (void)setShouldObserveDBModifications:(BOOL)shouldObserveDBModifications
             CGPoint newContentOffset = CGPointMake(0, MAX(0, newContentHeight - viewTopToContentBottom));
             [self.collectionView setContentOffset:newContentOffset animated:NO];
         }
+
+        // When we resume observing database changes, we want to scroll to show the user
+        // any new items inserted while we were not observing.  We therefore find the
+        // first item at or after the "view horizon".  See the comments below which explain
+        // the "view horizon".
+        ConversationViewItem *_Nullable lastViewItem = self.viewItems.lastObject;
+        BOOL hasAddedNewItems = (lastViewItem &&
+            previousLastTimestamp &&
+                                 lastViewItem.interaction.timestamp > previousLastTimestamp.unsignedLongLongValue);
+
+        DDLogInfo(@"%@ hasAddedNewItems: %d", self.logTag, hasAddedNewItems);
+        if (hasAddedNewItems) {
+            NSIndexPath *_Nullable indexPathToShow = [self firstIndexPathAtViewHorizonTimestamp];
+            if (indexPathToShow) {
+                [self.collectionView scrollToItemAtIndexPath:indexPathToShow
+                                            atScrollPosition:UICollectionViewScrollPositionCenteredVertically
+                                                    animated:YES];
```

Comment:
ntit: This happens before the view is visible, in viewWillAppear, so don't we want `animated:NO`?

Reply:
Reasonable.  I was thinking it'd be better to emphasize that we scrolled to the user, but ... it's not clear what's best.